charon/pkgs/radas_signature_handler.py

```diff
@@ -0,0 +1,209 @@
+from charon.constants import DEFAULT_RADAS_SIGN_TIMEOUT_RETRY_COUNT
+from charon.constants import DEFAULT_RADAS_SIGN_TIMEOUT_RETRY_INTERVAL
+from charon.pkgs.oras_client import OrasClient
+from proton import Event
+from proton.handlers import MessagingHandler
+
+logger = logging.getLogger(__name__)
+
+
+class UmbListener(MessagingHandler):
+    """
+    UmbListener class (AMQP version), register this when setup UmbClient
+    Attributes:
+        sign_result_loc (str): Local save path (e.g. “/tmp/sign”) for oras pull result,
+        this value transfers from the cmd flag, should register UmbListener when the client starts
+    """
+
+    def __init__(self, sign_result_loc: str) -> None:
+        super().__init__()
+        self.sign_result_loc = sign_result_loc
+
+    def on_start(self, event: Event) -> None:
+        """
+        On start callback
+        """
+        conf = get_config()
+        if not (conf and conf.is_radas_enabled()):
+            sys.exit(1)
+
+        rconf = conf.get_radas_config()
+        # explicit check to pass the type checker
+        if rconf is None:
+            sys.exit(1)
+        conn = event.container.connect(rconf.umb_target())
+        event.container.create_receiver(conn, rconf.result_queue())
+        logger.info("Listening on %s, queue: %s", rconf.umb_target(), rconf.result_queue())
+
+    def on_message(self, event: Event) -> None:
+        """
+        On message callback
+        """
+        # handle response from radas in a thread
+        thread = threading.Thread(target=self._process_message, args=[event.message.body])
+        thread.start()
+
+    def on_connection_error(self, event: Event) -> None:
+        """
+        On connection error callback
+        """
+        logger.error("Received an error event:\n%s", event)
+
+    def on_disconnected(self, event: Event) -> None:
+        """
+        On disconnected callback
+        """
+        logger.error("Disconnected from AMQP broker.")
+
+    def _process_message(self, msg: Any) -> None:
+        """
+        Process a message received from UMB
+        Args:
+            msg: The message body received
+        """
+        msg_dict = json.loads(msg)
+        result_reference_url = msg_dict.get("result_reference")
+
+        if not result_reference_url:
+            logger.warning("Not found result_reference in message，ignore.")
+            return
+
+        logger.info("Using SIGN RESULT LOC: %s", self.sign_result_loc)
+        sign_result_parent_dir = os.path.dirname(self.sign_result_loc)
+        os.makedirs(sign_result_parent_dir, exist_ok=True)
+
+        oras_client = OrasClient()
+        files = oras_client.pull(
+            result_reference_url=result_reference_url, sign_result_loc=self.sign_result_loc
+        )
+        logger.info("Number of files pulled: %d, path: %s", len(files), files[0])
+
+
+def generate_radas_sign(top_level: str, sign_result_loc: str) -> Tuple[List[str], List[str]]:
```

Comment:
Do you think we should add some utests for this method to make sure it works as expectation?

Reply:
@ligangty Yeah, I want to add the unit test for this feat, sign result parse and generation, etc, might need some mock if more thorough , I could file a unit test JIRA for this, WDYT?